src/chapter02/exercises/exercise03.lean

```diff
@@ -7,13 +7,31 @@ def rational (x : ℝ) :=
 /-- The product of two rational numbers is always rational. -/
 lemma part_a : ∀ {a b : ℝ}, rational a → rational b → rational (a * b) :=
 begin
-  sorry
+  intros a b,
+  rintro ⟨x1,y1,h1⟩ ⟨x2,y2,h2⟩,
+  use [x1*x2, y1*y2],
+  simp,
+  ring_nf,
+  rw [mul_inv, h1, h2],
+  ring_nf,
 end
 
 /-- The product of two irrational numbers is not always irrational. -/
 lemma part_b : ¬ ∀ {a b : ℝ}, irrational a → irrational b → irrational (a * b) :=
 begin
-  sorry
+  simp, 
+  use real.sqrt 2, 
+  split,
+  {exact irrational_sqrt_two,},
+  {use 1/(real.sqrt 2),
```

Comment:
Oh wow, why do you make it more complicated for yourself! Why not use sqrt 2 again? By the way if you `open real` you can just write `sqrt 2` instead of `real.sqrt 2`.